go.mod

```diff
@@ -8,13 +8,14 @@ require (
 	github.com/multiformats/go-multihash v0.0.14
 	github.com/multiformats/go-varint v0.0.6
 	github.com/stretchr/testify v1.7.0
-	golang.org/x/exp v0.0.0-20230626212559-97b1e661b5df
+	golang.org/x/exp v0.0.0-20230725093048-515e97ebf090
 )
 
 require (
 	github.com/davecgh/go-spew v1.1.0 // indirect
+	github.com/klauspost/cpuid/v2 v2.2.3 // indirect
 	github.com/minio/blake2b-simd v0.0.0-20160723061019-3f5f724cb5b1 // indirect
-	github.com/minio/sha256-simd v0.1.1-0.20190913151208-6de447530771 // indirect
+	github.com/minio/sha256-simd v1.0.1 // indirect
```

Comment:
Any reason to update this dependency? (Same for cupid as well)

Reply:
I believe those were updated by go mod tidy as dependencies of the sha package.

Reply:
Ah: I updated the sha package to fix a test failure due to a bug in the amd64 simd instructions. The newer version works and the older fails with a fatal error I've never seen before about the stack failing to unwind (or something like that - off the top of my head.)

Reply:
Can you please open a separate PR for any other dependency updates? Trying to keep the changes here minimal.

Reply:
Would be cool to fix this `golang.org/x/exp` package problem soon... because it breaks my app :(

Reply:
Or would it make sense to use the sort.Slice function instead until go 1.21 is released, even if that allocates memory compared to slices.Sort?

Reply:
Allocating is not an option, this function is getting called a lot. Should be easy enough to clean up this PR though. @paralin, are you planning to make these changes?

Reply:
+1 please merge this, we cannot update `/x/exp` package until this is fixed.

Reply:
Fixed.